wallets/nwc/index.js

```diff
@@ -33,61 +33,38 @@ export const card = {
   subtitle: 'use Nostr Wallet Connect for payments'
 }
 
-export async function nwcCall ({ nwcUrl, method, params }, { logger, timeout } = {}) {
-  const { relayUrl, walletPubkey, secret } = parseNwcUrl(nwcUrl)
-
-  const relay = await Relay.connect(relayUrl, { timeout })
-  logger?.ok(`connected to ${relayUrl}`)
+export async function getNwc (nwcUrl, { timeout = 5e4 } = {}) {
+  const ndk = Nostr.ndk
+  const { walletPubkey, secret, relayUrls } = parseNwcUrl(nwcUrl)
+  const nwc = new NDKNwc({
+    ndk,
+    pubkey: walletPubkey,
+    relayUrls,
+    secret
+  })
+  await nwc.blockUntilReady(timeout)
+  return nwc
+}
 
+/**
+ * Run a nwc function and throw if it errors
+ * (workaround to handle ambiguous NDK error handling)
+ * @param {function} fun - the nwc function to run
+ * @returns - the result of the nwc function
+ */
+export async function nwcTryRun (fun) {
   try {
-    const payload = { method, params }
-    const encrypted = await nip04.encrypt(secret, walletPubkey, JSON.stringify(payload))
-
-    const request = finalizeEvent({
-      kind: 23194,
-      created_at: Math.floor(Date.now() / 1000),
-      tags: [['p', walletPubkey]],
-      content: encrypted
-    }, secret)
-
-    // we need to subscribe to the response before publishing the request
-    // since NWC events are ephemeral (20000 <= kind < 30000)
-    const subscription = relay.fetch([{
-      kinds: [23195],
-      authors: [walletPubkey],
-      '#e': [request.id]
-    }], { timeout })
-
-    await relay.publish(request, { timeout })
-
-    logger?.info(`published ${method} request`)
-
-    logger?.info(`waiting for ${method} response ...`)
-
-    const [response] = await subscription
-
-    if (!response) {
-      throw new Error(`no ${method} response`)
-    }
-
-    logger?.ok(`${method} response received`)
-
-    if (!verifyEvent(response)) throw new Error(`invalid ${method} response: failed to verify`)
-
-    const decrypted = await nip04.decrypt(secret, walletPubkey, response.content)
-    const content = JSON.parse(decrypted)
-
-    if (content.error) throw new Error(content.error.message)
-    if (content.result) return content.result
-
-    throw new Error(`invalid ${method} response: missing error or result`)
-  } finally {
-    relay?.close()
-    logger?.info(`closed connection to ${relayUrl}`)
+    const { error, result } = await fun()
+    if (error) throw new Error(error.code + ' ' + error.message)
+    return result
+  } catch (e) {
+    if (e.error) throw new Error(e.error.code + ' ' + e.error.message)
+    throw e
```

Comment:
I think we should only log the human readable message and use the code as fallback if it's not set for some reason:

```suggestion
    const { error, result } = await fun()
    if (error) throw new Error(error.message || error.code)
    return result
  } catch (e) {
    if (e.error) throw new Error(e.error.message || e.error.code)
    throw e
```